### DISCUSSION TITLE
NO CHARGE Mark After Hit Charge

### DISCUSSION THREAD

**Discussion by ljmreal** [artisan-roaster-scope/artisan#1121] [General]
Hi there,

Need help, when I hit Charge. It showed no Charge. But the real problem, I don't see my BT progression. I can see the graphs of ROR, RORET, and ET.. I almost practically I am roasting blind on the most important graph.

But after the whole roast the BT shows up on the Artisan Viewer. If you use same roast recording as a background profile, it shows the BT.

I used my other notebook pc, it records and shows the BT during roasting.

Hoping to hear from you.

Lyndon

**Comment by poundy** [comment]
So, unless there's a screenshot or picture of the event, my first comment would be that there's not enough to try to track down what the behaviour might be. It's clear that the recording was running, but perhaps you were looking at the Artisan Viewer window instead? I've seen situations where as my first roast of the day Artisan has loaded an old profile into Viewer, and it's been in front of the active roasting window.

The simplest way (for you) to investigate this further is simply try to reproduce this, without actually roasting any coffee. Prove it happens again repeatably, and capture it happening in video from your phone

Then, show us all your evidence. Including supplying your config of the "working" and "failing" PCs

**Reply by ljmreal** [reply]
![Screenshot Viewer](https://user-images.githubusercontent.com/85690937/223593896-a25eaa04-a968-4699-825d-c8caeac733fd.png)
![Screenshot Actual Roast](https://user-images.githubusercontent.com/85690937/223593937-89b0940c-f179-4483-9bde-b26f4396974f.png)

Hi Poundy,

Please find uploaded screen shots..I am new in the forum..So I am not sure how to upload some image files.

Many thanks for your reply. Looking forward..

**Reply by poundy** [reply]
you got the upload of these two screenshots right, great !

now, what are we looking at? 

Clearly these just look like profiles - because they are. The Artisan Viewer window looks like you've adjusted something (the asterisk telling there's an unsaved change) and the main Artisan window has no pending save - but both are showing the same file that's just been loaded.  First up, an unsaved edit will cause issues with some actions (like auto open in Viewer at end of roast), so make sure you clean that up and then exit both Artisan and Viewer to ensure the state is flushed correctly.  I also assume these images are not "as it happened" footage but you've gone back to try to demonstrate what it looked like.  Unfortunately, that's not going to be that helpful...  What I really need you to do is to be able to demonstrate how you, and therefor we, can reproduce this fault now, or if this was just one-off issue to ignore.  If you can't reproduce it, then it's not actionable. If you can repro it, it can be explored, someone else can try to repro it with your settings etc to help isolate what is going on.

But... I note you're only using v2.6.0. Version 2.8.2 is the current released version, see https://github.com/artisan-roaster-scope/artisan/releases. We can't really do any sensible troubleshooting on this old version, the code has moved on, please update your install to use the latest, and then try to reproduce the issue (I'd say before upgrade, test if you can or can't repro it again, then test to see if that repro still works on the latest version).  This behaviour you're seeing could have actually been a weird bug that has been evident for some situation - I can't recall it, but again, no point troubleshooting, just jump forward and work on the latest codebase.

The last point I made was that your settings on the two PCs will be important to compare. Please save a copy of each of them, named appropriately so we can know working and not-working, and add a ".txt" extension on the end and then drag and drop the two txt files onto a reply here.  

But first and foremost, if you can't repro this, we can't, and there's no point investigating deeper, chasing a ghost. Let us know if you can repro it, then move to 2.8.2 and test again.

**Reply by poundy** [reply]
I've just noted ONE thing here. This particular trace MAY just have a problem that there's no lead in data and the formula that puts a CHARGE label on the graph is pushed too far left on Artisan but not on Viewer. Look at the data and confirm if a CHARGE event was registered. Ine thing I would say is that whatever your practice, you should always have lead-in data recorded. That means, for me, I always try to turn Artisan into ON mode at least 20 secs before charge to get a good 5+ data points active. But that doesn't go any way to explain why BT trace wouldn't have happened....

**Comment by ljmreal** [comment]
Hi There,

Actually, it was 2.8 version when this happened.
But my other laptop has the 2.6 version.. I tried to put back to 2.6 if it
will solve the problem, but that is not the case.

So what is your recommendation?

Kind regards,

Lyndon

**Reply by MAKOMO** [reply]
Do you have the BT flag ticked under menu `Config >> Device`, first tab `ET/BT` ?

<img width="946" alt="Screenshot 2023-03-08 at 06 40 41" src="https://user-images.githubusercontent.com/13909011/223628950-4a7179a4-8fb0-48f6-a30e-53462f223265.png">

**Reply by bigl00z3** [reply]
Marko,
I have observed a similar behavior but had not found the time yet to document it properly to fill a bug report. As I see this thread, I thought it may be worth mentionning (if there is any link to this issue... maybe not)

I go the feeling that it was when using the wrong set of artisan-settings. 
In my case, I had modified my settings and at one point I loaded old settings and opened a few old graphs. 
When I started artisan with recent profiles. BT was not showing but when opening files, it would display the AUC area but without the BT line. 
Ticking the LCDs did not show the curves and when looking in Device assignment everything was correct.

Once I reloaded the recent settings everything was back in order.

The problem is that I can't remember which settings and files I had used, otherwise I would have documented it.
I will try to reproduce and if I manage I will fill a bug report. 
But in the meantime I thought it was worth mentionning.

Nico

**Reply by MAKOMO** [reply]
Interesting. If possible, could you please attach your log file here to check for exceptions.

**Reply by MAKOMO** [reply]
@ljmreal: Could you please export your Artisan settings (menu Help >> Save Settings) as artisan-settings.txt and the profile you are loading as profile.txt by drag-and-drop here on the Github Web comment field. Note that you have to rename those files before to end in .txt. Thanks!

**Reply by bigl00z3** [reply]
@MAKOMO here is the log when trying to open an old file with different settings:
`03-08 12:00:46 MainThread artisanlib.main      INFO   : Artisan v2.8.0 (682bdaf, 0)
03-08 12:00:46 MainThread artisanlib.main      INFO   : locale: en
03-08 12:00:46 MainThread artisanlib.util      INFO   : debug logging OFF
03-08 12:00:49 MainThread artisanlib.main      ERROR  : At least one value in the dash list must be positive
Traceback (most recent call last):
  File "artisanlib\main.py", line 2981, in doUpdate
  File "matplotlib\backends\backend_agg.py", line 408, in draw
  File "matplotlib\artist.py", line 74, in draw_wrapper
  File "matplotlib\artist.py", line 51, in draw_wrapper
  File "matplotlib\figure.py", line 3074, in draw
  File "matplotlib\image.py", line 131, in _draw_list_compositing_images
  File "matplotlib\artist.py", line 51, in draw_wrapper
  File "matplotlib\axes\_base.py", line 3107, in draw
  File "matplotlib\image.py", line 131, in _draw_list_compositing_images
  File "matplotlib\artist.py", line 51, in draw_wrapper
  File "matplotlib\lines.py", line 799, in draw
  File "matplotlib\backend_bases.py", line 926, in set_dashes
ValueError: At least one value in the dash list must be positive
03-08 12:00:49 MainThread artisanlib.main      INFO   : profile loaded: C:/Users/Roaster/Batch1.alog
03-08 12:00:49 MainThread artisanlib.main      INFO   : startup time: 3.62
03-08 12:00:49 MainThread artisanlib.main      INFO   : phidgetLog started
03-08 12:00:49 MainThread artisanlib.main      INFO   : phidgetManager started
03-08 12:00:50 MainThread artisanlib.main      ERROR  : At least one value in the dash list must be positive
Traceback (most recent call last):
  File "artisanlib\main.py", line 2981, in doUpdate
  File "matplotlib\backends\backend_agg.py", line 408, in draw
  File "matplotlib\artist.py", line 74, in draw_wrapper
  File "matplotlib\artist.py", line 51, in draw_wrapper
  File "matplotlib\figure.py", line 3074, in draw
  File "matplotlib\image.py", line 131, in _draw_list_compositing_images
  File "matplotlib\artist.py", line 51, in draw_wrapper
  File "matplotlib\axes\_base.py", line 3107, in draw
  File "matplotlib\image.py", line 131, in _draw_list_compositing_images
  File "matplotlib\artist.py", line 51, in draw_wrapper
  File "matplotlib\lines.py", line 799, in draw
  File "matplotlib\backend_bases.py", line 926, in set_dashes
ValueError: At least one value in the dash list must be positive
`

I hope this can help

Nico

**Reply by MAKOMO** [reply]
Looking again at the screenshots above I think his issue is different from yours Nico. In his second (Artisan 2.6) screenshot you see that the BT is missing from the legend. This tells me that the BT flag is not ticked under Curves. If it would be clicked it would show the BT entry also in the legend. If the BT is hidden, event annotations are not drawn.

@bigl00z3 : Your issue is clearly a bug and I am investigating based on your information.

**Reply by ljmreal** [reply]
Hi Marko,

What I am wondering is even if you click the CHARGE bottom, the Event Annotation of CHARGE is not shown on the graph. But I believe the BT reminds hidden, cause it shows on the BT graph is shown later in the VIEWER.

My Curves has both ET and BT ticked. I will try to do a few roast today. And send you the profile.txt . Now my notebook pc is with 2.8.2 version (originally running at 2.8 version)

**Reply by roasterdave** [reply]
Lyndon, be sure to follow Marko's instructions to verify the BT curve is checked in Artisan before you start recording.  The Viewer's settings are completely independent.  

Because CHARGE is shown on the BT curve, so if the BT curve is not displayed the CHARGE event will not be displayed either.  CHARGE will be recorded in your profile regardless.  You can see this in your screenshot from the Viewer.

**Reply by ljmreal** [reply]
many thanks for your assistance. Grateful indeed.

I was looking at curves but not under the config>>devices which I seldom go into.

It is working now.

**Reply by poundy** [reply]
I would just like to point out that much earlier I requested you provide us the config files.

>> The last point I made was that your settings on the two PCs will be important to compare. Please save a copy of each of them, named appropriately so we can know working and not-working, and add a ".txt" extension on the end and then drag and drop the two txt files onto a reply here. 

Now, I would also like to remind you that Artisan is operated on a donation basis. If you're using it commercially to produce your coffee, it would be good karma to ensure you have made a donation to help support the work the team do.

**Comment by ljmreal** [comment]
Hi Marko,

Please find attached screenshots and profile log.
It just don't show the Charge, DE, FC annotation on the Artisan Screen..

Even if I hit Charge, the App does not show it Live during roasting. I am
relying on the ROR and ET.

I would be glad to receive your help. Meantime, I will be using my other
work notebook Pc to do my next roast.

Best regards,

Lyndon Realubit


[image: Brz SaoPao Viewer.png]

**Reply by MAKOMO** [reply]
Dear @ljmreal,

nothing attached to your message. Note that you need to go to the Web page with the discussion here on Github and do a drag-and-drop. Note further that I need both files, the profile as .alog Artisan profile (renamed to end in .txt before drag-and-drop) as well as your Artisan settings file as artisan-settings.aset.txt (again renamed to end in .txt otherwise the drag-and-drop does not work). You can export your settings via menu Help >> Save Settings to a file artisan-settings.aset.

**Reply by ljmreal** [reply]
[23-03-09_1051 Brz SnPao 9 Mar 23.txt](https://github.com/artisan-roaster-scope/artisan/files/10928365/23-03-09_1051.Brz.SnPao.9.Mar.23.txt)

[BSC05 artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/10928378/BSC05.artisan-settings.txt)

Please find attached renamed  alog as .txt  and artisan-settings.txt

Thanks.

**Reply by roasterdave** [reply]
`Please find attached renamed alog as .txt and artisan-settings.txt`

The settings file you posted has BT **unchecked**, so BT does not display.  As Marko suggested earlier [here](https://github.com/artisan-roaster-scope/artisan/discussions/1121#discussioncomment-5237146) simply go to Config>> Devices>> ET/BT tab and check the box for BT.  
Done.  The BT curve will display when you are roasting.

**Reply by MAKOMO** [reply]
Thanks for uploading those two files. As I expected, the BT curve is just hidden. Just click on the BT LCD (the one directly below the timer LCD in the upper right corner) to show that curve again. Alternatively you can go to the Device Assignment dialog, first tab `ET/BT` and tick the `BT` flag.

**Comment by ljmreal** [comment]
<img width="968" alt="Screenshot Brz SaoPao" src="https://user-images.githubusercontent.com/85690937/223909424-26ffe0d5-99ab-4291-aa4a-d252d59d3e18.png">
<img width="958" alt="Brz SaoPao Viewer" src="https://user-images.githubusercontent.com/85690937/223909436-913c969d-2a56-490b-9b8d-098eb21530cf.png">


Please find events recorded Live during roasting. NO Charge event, No DE, No FC.

But you know at the background the event was recorded.  Because it shows on the upper left of the screen. Event Recorded (Time, Temp , and if it FC )

Looking forward to hearing from you guys.

Lyndon

**Comment by ljmreal** [comment]
Hi Marko,

I have noticed once I hit Charge.  The bottom says grey.. And you can't
undo it if you want the change the start of your Charge.. Which I can do
before. Same goes with DE and FC, Drop. It just stays like that.

Best regards,
Lyndon
